### PR TITLE
feat(lambda): timeout property to lambda function construct 

### LINKED ISSUE #16047: (aws-lambda): add timeout property to Function
The timeout constructor parameter is defined as:
>The function execution time (in seconds) after which Lambda terminates the function.

I propose that this value be exposed as an instance parameter with the same name so it can be used in creating alerts and dashboards by customers.

### Use Case

I want to create dashboards and alerts when my lambdas execution time gets close to the `timeout` value defined for it.

I often have my dashboards and alerts in separate stacks. So when I pass the function object into the other stack (or make use of `Function.fromFunctionArn`), I have no way of knowing what timeout was defined. 

### Proposed Solution

The duration provided for `timeout` in the Function constructor should be available as a property of that instance.

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by ben-eb**
Hello 👋 

We would like to keep the timeout in sync between CloudWatch and Lambda, to avoid false positives. Currently, we are writing code like the following:

```typescript
const myLambdaTimeout = cdk.Duration.seconds(30);

const myLambda = new Lambda.Function(this, 'my-lambda', {
  // ... rest of props
  timeout: myLambdaTimeout
});

// ... later in the stack
new CloudWatch.Alarm(this, 'my-alarm', {
  // ... rest of props
  threshold: myLambdaTimeout.toMilliseconds()
});
```

It would be really nice to be able to do `myLambda.timeout.toMilliseconds()` instead, as we can't currently validate that the timeout being passed in is from that particular lambda function.

This fixes #16047.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by ben-eb** [comment]
Hello,

I've updated the branch a few times due to it being out of date with the `master` branch. Is there anything else I can do to get this reviewed/merged?

Thanks. 🙂

**Comment by nija-at** [comment]
Hi @ben-eb,

You do not need to update the branch regularly, unless you need the new updates on master. Our automation will take care of doing so once approved.

File: packages/@aws-cdk/aws-lambda/lib/function.ts
Comment (ben-eb) [reply packages/@aws-cdk/aws-lambda/lib/function.ts:579]: For our needs, an optional value is OK because we could throw an exception if the timeout was not defined.

A non-optional value would be a better API but this value is not guaranteed to be in the CloudFormation template. As such, if the default were changed in the future it might lead to CDK reporting a timeout value different to the one deployed to Lambda.

However, I'm happy to make that change if you're OK with this caveat. 👍

Comment (nija-at) [reply packages/@aws-cdk/aws-lambda/lib/function.ts:579]: Ok, fair enough. Let's leave it as is.

**Review by nija-at — APPROVED** [review]

**Comment by ben-eb** [comment]
Hey @nija-at,

I've created a new pull request with the "allow edits from maintainers" option checked, as it is currently unsupported for organisation accounts: https://github.com/aws/aws-cdk/pull/17594

Thanks for your patience! 👍

**Pull request closed without merging**
